### ISSUE TITLE
Bosses are (still) not spawning randomly.

### ISSUE THREAD

**Issue by seefourr** [TheWinABagel/Zenith#133]
Windows
minecraft 1.20.1
zenith 1.1.6 (downloaded from modrinth)
no other mods except zenith and relevant dependencies (fakerlib, zenith attributes, fabric api). Latest versions of each for 1.20.1

Created a hard world, set time to night. Waited for a while, no bosses spawn on their own.

I saw this was a previously open issue that was closed, but there aren't any new commits since the issue being closed (and apparently the 1.1.6 version from Modrinth doesn't fix it either). The github changelog additionally does not list any 1.1.6 changes.

(also I don't know what an "arch" issue is lol)

another small issue i ran into is when spawning bosses manually using either the boss summoning item or commands, sometimes a seemingly normal spider would be spawned instead.
That and the bosses try to attack me when I'm in creative mode for whatever reason.

**Comment by TheWinABagel** [comment]
Arch issue was an architectury api issue, which i *thought* was why this was happening... strange

**Comment by seefourr** [comment]
Still happening in 1.1.7 but I did notice from your code that you have a temporary fix. The issue still persists; bosses don't spawn on their own even when using just zenith + dependencies

but bosses do spawn when using zenith + dependencies + architectury 9.1.13. Which is an OK workaround for now

**Comment by DonTomasDeLaVega** [comment]
Yeah, but it looks like it might make you game crash everytime a boss tries to spawn. 
`Caused by: java.lang.IllegalArgumentException: Modifier is already applied on this attribute!`
Full crash lg : https://mclo.gs/4NR8cH2

**Comment by seefourr** [comment]
I was having that issue too, though not every boss spawn, it seemed to be somewhat random. I found that deleting all the "miniboss" json files from the jar _seemed_ to stop the issue mostly. At the very least I remember my server crashed less or not because of this bug.

I figure it could be the alternate workaround method of spawning bosses might be interfering with how the mod spawns its own minibosses like the honeyed archer etc, perhaps trying to layer them on each other and hence the attempting to reapply modifiers causing the crash.

Or it could just be another mod conflict I wasn't aware of. It was hard to test either way from my testing as the crashes were rather inconsistent even when i seriously upped the boss spawn rate.

**Comment by Kavik420** [comment]
I'm having the same issue
Windows
Minecraft 1.20.1
Zenith 1.1.9 - 1.20.1 from Modrinth

**Comment by schz404** [comment]
@seefourr I also downgraded to Architectury 9.1.13 and my game is also crashing. Did you find any other workarounds besides deleting the miniboss json files? How is this workaround doing with your server now after 2 weeks? Have you experienced significantly less crashes?

**Comment by Hapunn** [comment]
Still hoping for a fix to this issue as well. I've set up the adventure module to have gems and affix items only drop from bosses to give them some rarity in my server but bosses don't spawn. Minibosses do.

**Comment by ghost** [comment]
@Hapunn I think TheWinABagel abandoned the project or something idk

**Comment by TheWinABagel** [comment]
Not abandoned, just on a hiatus. Had a lot of IRL stuff happen, and had little motivation to work on Zenith instead of other, more (personally) interesting projects.

**Comment by TheWinABagel** [comment]
That being said, does [this version](https://github.com/TheWinABagel/Zenith/raw/1.20/zenith-1.1.9-1.20.1-event-fix.jar) fix this issue? If it does, lmk and I'll upload it officially

**Comment by schz404** [comment]
@TheWinABagel My bad. I hope everything is going well for you. I'll test the fix you attached.

**Comment by Kavik420** [comment]
@TheWinABagel I don't think the issue is fixed. Should I be running the latest Arch API version or downgrade?

**Comment by jodycircles** [comment]
@TheWinABagel The fix you attached seems to be working for me. I'll come back and post again if I run into any issues but for now bosses are spawning and haven't had any crashes. For anyone wondering I'm running AOF 7 and all I done was disable the Non-Patched Zenith and Enable the fix.

**Comment by schz404** [comment]
@TheWinABagel was out of town for a few days but I can now also confirm the fix works. Thank you!

**Comment by schz404** [comment]
I tried testing the fix really fast before I left, so I guess I just didn't wait long enough lol. I deleted my previous comment to avoid confusion for others.

**Comment by xXDogMilkXx** [comment]
It works on single player but doesnt work on a server

Or not, did fresh install and all works so far.

**Comment by bookmater** [comment]
Even using the event fixed version linked here I still cannot get the invading bosses to spawn, to make them spawn I set the spawn timer to 120 ticks and chance to 1.0 and stil nothing.
